searchlib/src/vespa/searchlib/common/bitvector.h

```diff
@@ -285,12 +287,12 @@ class BitVector : protected BitWord
     void setGuardBit() { setBit(size()); }
     void incNumBits() {
         if ( isValidCount() ) {
-            _numTrueBits++;
+            _numTrueBits.fetch_add(1, std::memory_order_relaxed);
         }
     }
     void decNumBits() {
         if ( isValidCount() ) {
-            _numTrueBits--;
+            _numTrueBits.fetch_sub(1, std::memory_order_relaxed);
```

Comment:
Is `incNumBits`/`decNumBits` possibly called from multiple threads concurrently? If not, and if visibility of changes to `_numTrueBits` are guaranteed between writer threads through other means (mutexes etc), this could be done without requiring explicit atomic bus operations. E.g.
```c++
_numTrueBits.store(_numTrueBits.load(std::memory_order_relaxed) - 1, std::memory_order_relaxed);
```

Reply:
Correct, there is only one writer thread anyway.

Reply:
Not very beautiful though....